backend/api/Services/ActivityService.cs

```diff
@@ -0,0 +1,122 @@
+using Pims.Dal.Entities;
+using Pims.Dal.Helpers.Extensions;
+using Pims.Dal.Repositories;
+using Pims.Dal.Security;
+
+namespace Pims.Api.Services
+{
+    public class ActivityService : BaseService, IActivityService
+    {
+        private readonly ILogger _logger;
+        private readonly IActivityRepository _activityRepository;
+        private readonly IActivityTemplateRepository _activityTemplateRepository;
+        private readonly IDocumentService _documentService;
+        private readonly INoteService _noteService;
+
+        public ActivityService(
+            ClaimsPrincipal user,
+            ILogger<ActivityService> logger,
+            IActivityRepository activityRepository,
+            IActivityTemplateRepository activityTemplateRepository,
+            INoteService noteService,
+            IDocumentService documentService)
+            : base(user, logger)
+        {
+            _logger = logger;
+            _activityRepository = activityRepository;
+            _activityTemplateRepository = activityTemplateRepository;
+            _noteService = noteService;
+            _documentService = documentService;
+        }
+
+        public PimsActivityInstance GetById(long id)
+        {
+            _logger.LogInformation("Getting activity {id}", id);
+            this.User.ThrowIfNotAuthorized(Permissions.NoteView);
+
+            // _user.ThrowIfNotAuthorized(Permissions.ResearchFileView);
+            var activityInstance = _activityRepository.GetById(id);
+
+            return activityInstance;
+        }
+
+        public IList<PimsActivityInstance> GetAllByResearchFileId(long researchFileId)
+        {
+            _logger.LogInformation("Getting activities by research id {researchFileId}", researchFileId);
+            this.User.ThrowIfNotAuthorized(Permissions.NoteView);
+
+            // _user.ThrowIfNotAuthorized(Permissions.ResearchFileView);
+            var activityInstances = _activityRepository.GetAllByResearchFileId(researchFileId);
+
+            return activityInstances;
+        }
+
+        public IList<PimsActivityTemplate> GetAllActivityTemplates()
+        {
+            _logger.LogInformation("Getting activity templates");
+            this.User.ThrowIfNotAuthorized(Permissions.NoteView);
+
+            // _user.ThrowIfNotAuthorized(Permissions.ResearchFileView);
+            var activtyTemplates = _activityTemplateRepository.GetAllActivityTemplates();
+            return activtyTemplates;
+        }
+
+        public PimsActivityInstance Add(PimsActivityInstance instance)
+        {
+            _logger.LogInformation("Adding activity instance ...");
+            this.User.ThrowIfNotAuthorized(Permissions.NoteAdd);
+
+            // _user.ThrowIfNotAuthorized(Permissions.ResearchFileAdd);
+            var newActivityInstance = _activityRepository.Add(instance);
+            _activityRepository.CommitTransaction();
+            return _activityRepository.GetById(newActivityInstance.ActivityInstanceId);
```

Comment:
we should not need to do a secondary retrieval after the commit.

Reply:
If you don't do this the concurrency id won't get updated and subsequent requests will fail.